boards/arm/rpi_pico/rpi_pico.dts

```diff
@@ -68,3 +70,17 @@
 &gpio0 {
 	status = "okay";
 };
+
+&i2c0 {
+	clock-frequency = <I2C_BITRATE_STANDARD>;
+	status = "okay";
```

Comment:
I don't think this should be enabled by default. If I'm not wrong, peripherals should be disabled by default, and enabled in an overlay file (with the console UART and GPIO being the exceptions). But input from others on this topic is welcomed.

Reply:
But the rp2040 has two i2c peripherals, one might enable i2c to use one and unintentionally enable the other. It happened to me once on an STM32 board, where UART stopped working because an i2c peripheral using the same pin was enabled by default

Reply:
So what do you suggest? To remove those ref nodes from dts or to make status disabled for both of them?

Reply:
Applications can always override using an overlay, so if they want i2c1 and not i2c0, an overlay can be used:

```
&i2c0 {
    status = "disabled";
};
```

Reply:
True, but In my opinion it's better to have to explicitly enable peripherals than having to explicitly disable them. The latter case makes the user responsible for making sure peripherals are not accidentally enabled, which assumes that the user is aware of this problem. Accidentally turning on a peripheral can increase power consumption and cause conflicts like I had.

Reply:
@yonsch I agree, I left the nodes in dts and I put status disabled on them.

Reply:
@yonsch Can you take a look at boards/arm/nucleo_f091rc/nucleo_f091rc.dts? It is similar with what I did for Pico's I2C.

Reply:
Okay, if this is the policy I won't discuss it here, though I might continue this discussion somewhere else :) . Meanwhile, all my comments are resolved

Reply:
@yonsch sane defaults are certainly worth considering, but perhaps a broader topic. Can you please file an issue and mark it with dev-review perhaps if you want to discuss in a live setting. I generally agree that surprising defaults are a misfeature, but the way this is done here is similar to many other dts files in the tree today so certainly shouldn't be a blocker for this particular PR.

Reply:
I tend to agree with @yonsch. Here is our policy:

https://docs.zephyrproject.org/latest/guides/porting/board_porting.html#general-recommendations

In particular (emphasis mine):

> Unless explicitly recommended otherwise by this section, leave **peripherals and their drivers** disabled by default.

So @gmarull 's comment that the driver is enabled by default does not meet this requirement.

Reply:
>It's ok to have status = "okay";. Board doesn't set CONFIG_I2C=y, so i2c driver won't be enabled unless applications do.

@gmarull 's comment is correct, decisive is `CONFIG_I2C` and reflects our general practice, it is also consistent with board porting guide.

Reply:
If we let boards add `status = "disabled";`, it means that we'll need many overlays in samples that are board-specific, whereas `CONFIG_I2C=y` in `prj.conf` works for all boards that have an I2C enabled in DT. I think it's a bit confusing, but I can't think of a better way.

Reply:
Opened an issue to continue this discussion: #42998

Reply:
Note that this would likely not happen with a custom out-of-tree board. In those cases, you generally describe what you have/need, and the application tends to be specifically designed for that hardware. Upstream, we have lots of eval kits or demo boards that do not have a specific purpose. Or sometimes, boards are meant to be plugged into an expansion kit, where hardware really gets defined from a board perspective (see for example https://www.tindie.com/products/arturo182/rp2040-stamp/). For maintainability reasons, I'd avoid going the path of disabling peripherals and require sample overlays.